src/tokenizer.rs

```diff
@@ -447,12 +447,21 @@ impl<'a> State<'a> {
 pub struct Tokenizer<'a> {
     dialect: &'a dyn Dialect,
     query: &'a str,
+    options: &'a TokenizerOptions,
+}
+
+pub struct TokenizerOptions {
+    pub no_escape: bool,
 }
 
 impl<'a> Tokenizer<'a> {
     /// Create a new SQL tokenizer for the specified SQL statement
-    pub fn new(dialect: &'a dyn Dialect, query: &'a str) -> Self {
-        Self { dialect, query }
+    pub fn new(dialect: &'a dyn Dialect, query: &'a str, options: &'a TokenizerOptions) -> Self {
```

Comment:
Since this is one of the main interfaces to SqlParser, I think changing the signature will be a fairly large change. If we are going to do so I would prefer to combine the dialect and options into a single structure

```rust
pub struct TokenizerOptions<'a> {
    pub no_escape: bool,
    dialect: &'a dyn Dialect,
}
...

impl<'a> Tokenizer<'a> {
    pub fn new(options: TokenizerOptions<'a>, query: &'a str,) -> Self {
...
}
```

That way the API gets more general and we won't have to change it again if we add new parameters

Reply:
@alamb
> Since this is one of the main interfaces to SqlParser, I think changing the signature will be a fairly large change.

While I strongly agree with this idea, I think `dialect` and other options such as `no_escape` would be better to be separated. :)
There are two reasons for this.
1. If we separate `dialect` and other options(`TokenizerOptions`), you won't have to change the interface of `Tokenizer` even when you want to add new parameters. This is because all you have to do will be just adding the new option to `TokenizerOptions`. So the interface of `Tokenizer::new()` itself will not change, i think.
2. `Parser` has the below interface now, and `dialect` and `options` are separated here.
https://github.com/canalun/sqlparser-rs/blob/2e6502e40b9b777e0ee590e5aa172a3c1bc2a42d/src/parser.rs#L204
And this separation is assumed in the logic of `dialect_of`, which is called against both `Parser` and `Tokenizer`.
https://github.com/canalun/sqlparser-rs/blob/2e6502e40b9b777e0ee590e5aa172a3c1bc2a42d/src/dialect/mod.rs#L54
So we have to align the two interfaces of `Parser` and `Tokenizer`. And if point 1 is true, I think it would be better to set the same interface as `Parser` on `Tokenizer`.

How about it?? :)